agent/agent.go

```diff
@@ -781,10 +781,18 @@ func (a *agent) handleReconnectingPTY(ctx context.Context, msg codersdk.Reconnec
 			<-ctx.Done()
 			_ = process.Kill()
 		}()
+
+		lastRead := time.Now()
 		go func() {
-			// If the process dies randomly, we should
-			// close the pty.
+			// If the process exits, we should close the pty.
 			_ = process.Wait()
+			if time.Since(lastRead) < 2*time.Second {
```

Comment:
Is there any reason why it needs 2 seconds?

BTW It might be a good idea to arrange a dedicated `const` block for all timeouts defined in `agent.go`.